src/store/wallet/hooks.ts

```diff
@@ -2,71 +2,64 @@ import React, { useState, useEffect, useMemo } from 'react';
 
 import BigNumber from 'bignumber.js';
 import { isAddress } from 'icon-sdk-js/lib/data/Validator.js';
+import JSBI from 'jsbi';
 import _ from 'lodash';
 import { BalancedJs } from 'packages/BalancedJs';
-import IRC2 from 'packages/BalancedJs/contracts/IRC2';
-import ContractSettings from 'packages/BalancedJs/contractSettings';
 import { useIconReact } from 'packages/icon-react';
 import { useDispatch, useSelector } from 'react-redux';
 
 import bnJs from 'bnJs';
-import { NETWORK_ID } from 'constants/config';
 import { MINIMUM_ICX_FOR_TX } from 'constants/index';
+import { BIGINT_ZERO } from 'constants/misc';
 import { SUPPORTED_TOKENS_LIST, isNativeCurrency, SUPPORTED_TOKENS_MAP_BY_ADDRESS, isBALN } from 'constants/tokens';
 import { useBnJsContractQuery } from 'queries/utils';
 import { useAllTransactions } from 'store/transactions/hooks';
+import { useUserAddedTokens } from 'store/user/hooks';
 import { Token, CurrencyAmount, Currency } from 'types/balanced-sdk-core';
 import { Pair } from 'types/balanced-v1-sdk';
 
 import { AppState } from '..';
 import { useAllTokens } from '../../hooks/Tokens';
-import { changeBalances, resetBalances } from './actions';
-
-const contractSettings = new ContractSettings({ networkId: NETWORK_ID });
+import { changeBalances } from './actions';
 
 export function useWalletBalances(): AppState['wallet'] {
   return useSelector((state: AppState) => state.wallet);
 }
 
+export function useAvailableBalances(
+  account: string | undefined,
+  tokens: Token[],
+): {
+  [key: string]: CurrencyAmount<Currency>;
+} {
+  const balances = useCurrencyBalances(account || undefined, tokens);
+
+  return React.useMemo(() => {
+    return balances.reduce((acc, balance) => {
+      if (!balance) return acc;
+      if (!JSBI.greaterThan(balance.quotient, BIGINT_ZERO)) return acc;
+
+      acc[balance.currency.wrapped.address] = balance;
+
+      return acc;
+    }, {});
+  }, [balances]);
+}
+
 export function useWalletFetchBalances(account?: string | null) {
```

Comment:
After this refactoring, I thought that we can remove the wallet redux store. 
I think that we can `useWalletFetchBalances` hook and move the logic into the `useWalletBalances` hook.

Reply:
then we can remove this code.
https://github.com/balancednetwork/balanced-network-interface/blob/4a330afa77acfadc5590eb937f0f56e62da7a229/src/app/containers/TradePage/index.tsx#L22